src/librustc_passes/loops.rs

```diff
@@ -94,32 +102,41 @@ impl<'a, 'hir> Visitor<'hir> for CheckLoopVisitor<'a, 'hir> {
                     },
                     Err(hir::LoopIdError::UnresolvedLabel) => ast::DUMMY_NODE_ID,
                 };
+
                 if loop_id != ast::DUMMY_NODE_ID {
                     match self.hir_map.find(loop_id).unwrap() {
                         hir::map::NodeBlock(_) => return,
                         _=> (),
                     }
                 }
 
+                if self.cx == LabeledBlock {
+                    return;
+                }
```

Comment:
This should have at least a comment. `Context::LabeledBlock` seems dangerous, since breaks to labelled blocks should generally not interact with anything loop-related.